### DISCUSSION TITLE
Prevent user from saving an object

### DISCUSSION THREAD

**Discussion by plogramer** [sadieblue/agilityblue.workspace-scripts#5] [Q&A] [answered]
I would like to prevent users from saving object based on some custom rules.
For example, we have a custom text field that we would like users to follow specific format - i.e. [4-digit]-[4-digit]. 
If user doesn't follow the rule, I would like to pop-up a message and stop saving the entry.

**Comment by mrabb24** [comment, marked as answer]
Hello, 

If you use a "Before Save" action state on your event, you can prevent an object from being saved by returning `$null` if it doesn't mean your validation requirements. If it does satisfy your requirements, you return the inbound object.

Example:

```powershell
$mediaLogEntry = Get-InboundObjectInstance

# Define the regex pattern for [4-digit]-[4-digit] (adjust to your needs)
$regexPattern = '^\d{4}-\d{4}$'

# Check the field you want to match the pattern against
if (-not ($mediaLogEntry.Reference -match $regexPattern)) {
    Write-Output "The reference value `"$($mediaLogEntry.Reference)`" did not match the required regex pattern"
    
    # Return null here. Agility Blue will prevent the object from being saved and indicate
    # to the user that the object was not saved due to a script error.
    return $null
}

Write-Output "The reference value `"$($mediaLogEntry.Reference)`" matched the required regex pattern"

# Return the media log entry back to the pipeline so Agility Blue can continue on with
# saving it.
return $mediaLogEntry
```

You can't adjust the message returned at this time, but that is something we're working on allowing in a future update.

Matt

**Reply by plogramer** [reply]
Thank you! This is exactly what I was asking. Do you know when we are able to customize the output message?

**Reply by mrabb24** [reply]
Happy to hear that helped! We don't have any official dates to provide for the custom output message yet, but I can tell you it's actively being worked on.
